example/main.py

```diff
@@ -1,163 +1,142 @@
+    classes = ('plane', 'car', 'bird', 'cat', 'deer', 'dog', 'frog', 'horse', 'ship', 'truck')
     net.eval()
-    dataiter = iter(testloader)
-    images, labels = next(dataiter)
-
-    outputs = net(images)
-
-    _, predicted = torch.max(outputs, 1)
-
-    correct = 0
-    total = 0
+    running_loss = 0.0
+    
     with torch.no_grad():
         for data in testloader:
             images, labels = data
-            outputs = net(images)
-            _, predicted = torch.max(outputs.data, 1)
-            total += labels.size(0)
-            correct += (predicted == labels).sum().item()
 
-    print('Accuracy of the network on the 10000 test images: %d %%' % (
-        100 * correct / total))
+            if args.cuda:
+                images, labels = images.cuda(), labels.cuda()
 
-    class_correct = list(0. for i in range(10))
-    class_total = [0. for _ in range(10)]
-    with torch.no_grad():
-        for data in testloader:
-            images, labels = data
             outputs = net(images)
             _, predicted = torch.max(outputs, 1)
-            c = (predicted == labels).squeeze()
-            for i in range(4):
-                label = labels[i]
-                class_correct[label] += c[i].item()
-                class_total[label] += 1
+            running_loss += F.cross_entropy(outputs, labels).item()
 
-    for i in range(10):
-        print('Accuracy of %5s : %2d %%' % (
-            classes[i], 100 * class_correct[i] / class_total[i]))
+    print('Loss: {:.3f}'.format(running_loss / len(testloader)))
+    print('Accuracy: {:.3f}'.format(accuracy_score(predicted, labels)))
+    print(classification_report(predicted, labels, target_names=classes))
+    
+    return running_loss
 
+def init_server(args):
+    model = LeNet()
+    server = ParameterServer(model=model)
+    server.run()
 
-if __name__ == '__main__':
-    main()
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Distbelief training example')
+    parser.add_argument('--batch-size', type=int, default=64, metavar='N', help='input batch size for training (default: 64)')
+    parser.add_argument('--test-batch-size', type=int, default=10000, metavar='N', help='input batch size for testing (default: 1000)')
+    parser.add_argument('--epochs', type=int, default=10, metavar='N', help='number of epochs to train (default: 10)')
+    parser.add_argument('--lr', type=float, default=0.1, metavar='LR', help='learning rate (default: 0.1)')
+    parser.add_argument('--freq', type=int, default=10, metavar='N', help='how often to send/pull grads (default: 10)')
+    parser.add_argument('--cuda', action='store_true', default=False, help='use CUDA for training')
+    parser.add_argument('--log-interval', type=int, default=10, metavar='N', help='how many batches to wait before logging training status')
+    parser.add_argument('--distributed', action='store_true', default=False, help='whether to use DownpourSGD or normal SGD')
```

Comment:
should we have `--distributed` on by default?

Reply:
yes, but i didn't quite know the proper syntax for it

Reply:
probably `default=True`